vault/core.go

```diff
@@ -1114,10 +1129,25 @@ func (c *Core) sealInitCommon(req *logical.Request) (retErr error) {
 		return retErr
 	}
 
+	// Tell any requests that know about this to stop
+	if c.requestContextCancelFunc != nil {
+		c.requestContextCancelFunc()
+	}
+
+	// Unlock from the request handling
+	c.stateLock.RUnlock()
```

Comment:
You should probably unlock the statelock everytime this function returns above :(